### PR TITLE
Migrate Vulnerability Detector IT

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|---|
|#17403 |

## Description

This Issue aims to migrate the Vulnerability Detector IT suite from the QA repository.

## Tests
| path | Test Execution | OS |Commit | Notes |
|--|--|--|--|--|
| test_vulnerability_detector | [:green_circle:](https://github.com/wazuh/wazuh/files/12527837/VDT.Tests.zip)  |  Local Manager | https://github.com/wazuh/wazuh/pull/17576/commits/acbde76609a3d9d12e0f4e345ca3078e2a8d18b6 | Local results containing all tests. Tier 0 and 1 are run through Github actions.

> NOTES:
> This PR has been recreated with New branch. Original development history in https://github.com/wazuh/wazuh/pull/17576

### CODE REVIEW

**Review by TomasTurina — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                                               update_configuration_template)
+from wazuh_testing.utils.callbacks import generate_callback
+from wazuh_testing.modules.vulnerability_detector import patterns as cb
+from wazuh_testing.modules.vulnerability_detector.utils import VULNERABLE_PACKAGES_EMPTY_VENDOR
+from test_vulnerability_detector.utils import check_cve_affects_package_alert
+from wazuh_testing.tools.monitors.file_monitor import FileMonitor
+from . import TEST_CASES_PATH, CONFIGURATIONS_PATH, custom_nvd_json_feed_path
+
+
+local_internal_options = {'wazuh_modules.debug': '2', 'monitord.rotate_log': '0'}
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/09cbf3c1f5304969cef921b94d4eb54d55450926

File: tests/integration/test_vulnerability_detector/test_cpe_helper/data/configuration_template/configuration_cpe_indexing.yaml
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/data/configuration_template/configuration_cpe_indexing.yaml:1]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/19f54949179d6b93a1f0a543505cf5e4531dffa2

File: tests/integration/test_vulnerability_detector/test_general_settings/test_min_full_scan_interval.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                                             configuration_metadata)
+systems = [metadata['system'] for metadata in configuration_metadata]
+
+# Set offline custom feeds configuration
+to_modify = ['RHEL_FEED_PATH', 'custom_rhel_json_feed_path', 'custom_nvd_json_feed_path']
+new_values = [custom_rhel_oval_feed_path, custom_rhel_json_feed_path, custom_nvd_json_feed_path]
+configurations = update_configuration_template(configurations, to_modify, new_values)
+
+
+@pytest.fixture()
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_general_settings/test_min_full_scan_interval.py]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/37a1763210fa94466a038d4cdc53cff62e5f898e

File: tests/integration/test_vulnerability_detector/test_cpe_helper/data/test_cases/cases_cpe_indexing_missing_fields.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    NVD_JSON_PATH: CUSTOM_NVD_JSON_PATH
+  metadata:
+    system: WINDOWS_10
+    wrong_field: null
+    missing_field:
+      - ACTION_TAG
+    expected_result: package_not_indexed
+    tags: []
+    values: []
+- name: missing_vendor_field
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/data/test_cases/cases_cpe_indexing_missing_fields.yaml:68]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/967b5f8705ac6c377d0c1450cff1b4533107691c

File: .github/workflows/integration-tests-vulnerability_detector-tier-2.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+          echo "" >> ./etc/preloaded-vars.conf
+          echo 'USER_ENABLE_AUTHD="n"' >> ./etc/preloaded-vars.conf
+          echo "" >> ./etc/preloaded-vars.conf
+          echo 'USER_AUTO_START="y"' >> ./etc/preloaded-vars.conf
+          echo "" >> ./etc/preloaded-vars.conf
+          sudo sh install.sh
+          rm ./etc/preloaded-vars.conf
+      # Download and install integration tests framework.
+      - name: Download and install integration tests framework
+        run: |
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply .github/workflows/integration-tests-vulnerability_detector-tier-2.yml:60]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/967b5f8705ac6c377d0c1450cff1b4533107691c

File: .github/workflows/integration-tests-vulnerability_detector-tier-2.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+name: Integration tests for Vulnerability Detector - Tier 2
+
+on:
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply .github/workflows/integration-tests-vulnerability_detector-tier-2.yml:4]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/967b5f8705ac6c377d0c1450cff1b4533107691c

File: .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml [hunk 6]
```diff
@@ -0,0 +1,73 @@
+name: Integration tests for Vulnerability Detector - Tier 0 and 1
+
+on:
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml:4]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/967b5f8705ac6c377d0c1450cff1b4533107691c

File: .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          echo "" >> ./etc/preloaded-vars.conf
+          echo 'USER_ENABLE_AUTHD="n"' >> ./etc/preloaded-vars.conf
+          echo "" >> ./etc/preloaded-vars.conf
+          echo 'USER_AUTO_START="y"' >> ./etc/preloaded-vars.conf
+          echo "" >> ./etc/preloaded-vars.conf
+          sudo sh install.sh
+          rm ./etc/preloaded-vars.conf
+      # Download and install integration tests framework.
+      - name: Download and install integration tests framework
+        run: |
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml:68]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/967b5f8705ac6c377d0c1450cff1b4533107691c

File: tests/integration/conftest.py [hunk 8]
```diff
@@ -277,6 +277,15 @@ def daemons_handler_module(request: pytest.FixtureRequest) -> None:
     yield from daemons_handler_implementation(request)
 
 
+@pytest.fixture(scope='module')
+def restart_wazuh_daemon_after_finishing_module(daemon: str = None) -> None:
```
Comment (TomasTurina) [inline tests/integration/conftest.py:281]: This is only used for VD, move it to `tests/integration/test_vulnerability_detector/conftest.py`. Add a better description.

**Review by TomasTurina — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TomasTurina) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py]: Fix this after apply https://github.com/wazuh/qa-integration-framework/pull/14#discussion_r1315912175.

File: tests/integration/test_vulnerability_detector/conftest.py [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+                                                                insert_suse_system_package, VULNERABLE_PACKAGES)
+from wazuh_testing.utils.db_queries import agent_db
+from wazuh_testing.utils.time import get_current_timestamp
+
+
+CUSTOM_CPE_HELPER_PATH = str(Path(FEEDS_PATH, 'cpe_helper', CUSTOM_CPE_HELPER))
+
+
+@pytest.fixture()
+def restart_modulesd() -> None:
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/conftest.py]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/a3d84c65baa15e243002cd4e17de2c6512a2809c

File: tests/integration/conftest.py (hunk 8 above)
Comment (Deblintrake09) [reply tests/integration/conftest.py:281]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/a9d46de37527cd7de57335f479d99d19bff3c6d7

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py]: Done in https://github.com/wazuh/wazuh/pull/18567/commits/03529a6908a2df4408a20e472d844dea56b10931

**Pull request merged**